### ISSUE TITLE
/bin/bash: line 4: tesseract: command not found 

### ISSUE THREAD

**Issue by toobluescientist** [nicooprat/alfred-ocr#12]
when I run the workflow, the pasted clipboard appears this: `/bin/bash: line 4: tesseract: command not found`

how to fix it? 

I ran tesseract in Terminal and it worked well but not in Alfred.

**Comment by nicooprat** [comment]
Hi, can you try to restart Alfred after installation?

**Comment by 1activegeek** [comment]
I'm running into this issue as well. I have it running on my old mac, but trying to install on my new Mac, I've hit a snag. I think it may be relative to how Alfred runs scripts and evaluates the PATH. You'll see here that I do in fact have tesseract installed and responding via its name on the command line, yet the script when run from Alfred, is not working. 

![Screen Shot 2021-05-05 at 12 01 37 PM](https://user-images.githubusercontent.com/6810720/117172884-3c1c7980-ad9a-11eb-8f48-741908a20761.png)

I did the following:
- Installed tesseract via homebrew, attempted removing and reinstalling - and then performing all steps below again as well
- Quit Alfred completely and restarted to ensure it had caught all updates
- Quit Terminal and restarted to ensure path was updated for all instances
- Attempted running tesseract command by itself (as seen in this screenshot)
- Tried altering the script to use /bin/zsh instead

Finally was able to make this work by manually adding the location of tesseract (via homebrew) to the path list in the Run Script portion of the workflow. So I modified line 1 `export PATH=/usr/local/bin/:$PATH` to `export PATH=/usr/local/bin/:/opt/homebrew/bin:$PATH`. If you don't know where your install is running from, run `which tesseract` from the command line to identify. I'm on a Mac here obviously, but you should be able to adapt to Linux as well. 

To Dev - I think this just means there needs to be a more robust way built in to identify the location of the tesseract binary if coming from the PATH variable.

**Comment by dsbekker** [comment]
Just to expand on this slightly. The issue affects apple silicon (M1) macs. 

Homebrew on ARM installs to /opt/ as opposed to /usr/ on intel macs.

**Comment by blindborges** [comment]
> I'm running into this issue as well. I have it running on my old mac, but trying to install on my new Mac, I've hit a snag. I think it may be relative to how Alfred runs scripts and evaluates the PATH. You'll see here that I do in fact have tesseract installed and responding via its name on the command line, yet the script when run from Alfred, is not working.
> 
> ![Screen Shot 2021-05-05 at 12 01 37 PM](https://user-images.githubusercontent.com/6810720/117172884-3c1c7980-ad9a-11eb-8f48-741908a20761.png)
> 
> I did the following:
> 
> * Installed tesseract via homebrew, attempted removing and reinstalling - and then performing all steps below again as well
> * Quit Alfred completely and restarted to ensure it had caught all updates
> * Quit Terminal and restarted to ensure path was updated for all instances
> * Attempted running tesseract command by itself (as seen in this screenshot)
> * Tried altering the script to use /bin/zsh instead
> 
> Finally was able to make this work by manually adding the location of tesseract (via homebrew) to the path list in the Run Script portion of the workflow. So I modified line 1 `export PATH=/usr/local/bin/:$PATH` to `export PATH=/usr/local/bin/:/opt/homebrew/bin:$PATH`. If you don't know where your install is running from, run `which tesseract` from the command line to identify. I'm on a Mac here obviously, but you should be able to adapt to Linux as well.
> 
> To Dev - I think this just means there needs to be a more robust way built in to identify the location of the tesseract binary if coming from the PATH variable.



I'm stuck in the same problem.

I installed tesseract via macports because brew was not working.

And my tesseract is at "/opt/local/bin/tesseract"

I tried to fix export path as many as I could, but all of them didn't work.

Can you give me an advice?

**Comment by nicooprat** [comment]
Those path issues are really annoying. Could you try to use `/bin/zsh` instead of `/bin/bash` if you're using latest Mac version?

![image](https://user-images.githubusercontent.com/645641/151776870-4154ef29-6e25-4c33-b009-8dff3a94c7e2.png)

**Comment by blindborges** [comment]
Yes, it is really annoying. I thought I was so close, but It didn’t work.

I will try.

Thanks for replying.

**Comment by nicooprat** [comment]
Looks like this is a common issue in Alfred plugins, you can take a look here to see what's done in PHP :D 

https://github.com/tillkruss/alfred-lorem-ipsum/issues/6#issuecomment-1006756654

**Comment by blindborges** [comment]
Thanks for helping me out again.

I really appreciate it.



2022년 1월 31일 (월) 오후 10:49, Nico Prat ***@***.***>님이 작성:
